evergreen.yml

```diff
@@ -1582,6 +1582,47 @@ functions:
         permissions: public-read
         content_type: application/json
 
+  "static code analysis":
+    - command: shell.exec
+      type: test
+      params:
+        shell: bash
+        working_dir: mongosql-odbc-driver
+        script: |
+          ${prepare_shell}
+          export SEMGREP_APP_TOKEN=${semgrep_app_token}
+          echo "Running static code analysis with Semgrep..."
+          venv='venv'
```

Comment:
What's the benefit of this?

Reply:
ahhh, if you are really interested you can read at this thread: https://mongodb.slack.com/archives/C05N744KWQ4/p1717592697786179

Short explanation if you don't want to get bogged down with details : Semgrep has 2 different set of rules. Community rules and Pro rules. For Pro rules to be used, you need a Semgrep account and generate an API token.

For Rust, useful rules are only Pro rules. So, without an account, we had no rules applied to the scan (which yield a perfectly clean result of no vulnerabilities :P )

Reply:
I was specifically talking about `venv='venv'` not the `SEMGREP_APP_TOKEN`

Reply:
When I was making this script, I was largely just copying what had already be done in the BI-Connector and JDBC. This was done in both those repos (These PRs show it: https://github.com/10gen/sqlproxy/pull/453 and https://github.com/mongodb/mongo-jdbc-driver/pull/261), so I included it here, but I'm not actually sure what the point of it is. It doesn't seem necessary.

Reply:
I ran a patch without it, it seems to work fine: https://spruce.mongodb.com/version/666202b446b7c80007c8f008/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

new code:
```
          echo "Running static code analysis with Semgrep..."

          # Setup or use the existing virtualenv for semgrep
          if [[ -f "venv/bin/activate" ]]; then
              echo 'using existing virtualenv'
              . venv/bin/activate
          else
              echo 'Creating new virtualenv'
              python3 -m virtualenv venv
              echo 'Activating new virtualenv'
              . venv/bin/activate
          fi
          python3 -m pip install semgrep
```